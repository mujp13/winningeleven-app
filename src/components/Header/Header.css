.landing-page {
  font-family: "Bungee Inline", cursive;
  font-size: 25px;
  text-align: center;
}

.Header {
  height: 50px;
  display: flex;
  align-items: normal;
  justify-content: space-between;
}

.Header h1 {
  margin: 0;
}

.Header h1 a {
  text-decoration: none;
}

.Header a {
  color: #444;
}

.Header__not-logged-in {
  display: flex;
  flex-direction: column;
}

.Header__not-logged-in .Hyph {
  display: none;
}

@media screen and (min-width: 450px) {
  .Header {
    align-items: baseline;
  }
  .Header h1 {
    font-size: 22pt;
  }

  .Header__not-logged-in {
    flex-direction: row;
  }

  .Header__not-logged-in .Hyph {
    display: initial;
  }
}
