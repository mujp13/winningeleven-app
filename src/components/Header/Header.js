import React, { Component } from "react";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Hyph } from "../Utils/Utils";
import TokenService from "../../services/token-service";
import IdleService from "../../services/idle-service";
import "./Header.css";

export default class Header extends Component {
  handleLogoutClick = () => {
    TokenService.clearAuthToken();
    /* when logging out, clear the callbacks to the refresh api and idle auto logout */
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
  };

  renderLogoutLink() {
    return (
      <div className="Header__logged-in">
        <Link to="/newteam">New Team</Link>
        <Hyph />
        <Link onClick={this.handleLogoutClick} to="/">
          Logout
        </Link>
      </div>
    );
  }

  renderLoginLink() {
    return (
      <div className="Header__not-logged-in">
        <Link to="/register">Register</Link>
        <Hyph />
        <Link to="/login">Sign in</Link>
      </div>
    );
  }

  render() {
    return (
      <>
        <nav className="Header">
          <h1>
            <Link to="/">
              <FontAwesomeIcon className="black" icon="futbol" /> Winning Eleven
            </Link>
          </h1>
          {TokenService.hasAuthToken()
            ? this.renderLogoutLink()
            : this.renderLoginLink()}
        </nav>
        <p className="landing-page">
          You got what it takes to make the best team in the world? <br></br>{" "}
          Play our soccer fantasy game in which users build an imaginary team of
          real life players.
        </p>
      </>
    );
  }
}
