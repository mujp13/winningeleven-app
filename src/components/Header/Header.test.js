import React from "react";
import ReactDOM from "react-dom";
import Header from "./Header";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>,
    div
  );

  ReactDOM.unmountComponentAtNode(div);
});

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
