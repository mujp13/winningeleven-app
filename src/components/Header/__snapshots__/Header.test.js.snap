// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="Header"
  >
    <h1>
      <a
        href="/"
        onClick={[Function]}
      >
         Winning Eleven
      </a>
    </h1>
    <div
      className="Header__not-logged-in"
    >
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
      <span
        className="Hyph"
      >
         - 
      </span>
      <a
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </nav>,
  <p
    className="landing-page"
  >
    You got what it takes to make the best team in the world? 
    <br />
     
    Play our soccer fantasy game in which users build an imaginary team of real life players.
  </p>,
]
`;
